### PR TITLE
fix: back button now correctly refreshes route

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- looks like close handler for modal introduced issue with back button not working because `onCloseModal` changed history to interfere with Next.js router
- back button now works when you are navigating between multiple tag pages by constantly selecting them eg. go to /tags/php (select tag) -> /tags/javascript -> back -> /tags/php (before URL would change but the page would stay the same as if you are on javascript page)

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### Preview domain
https://fix-back-button-on-feed-pages.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/hooks/usePostModalNavigation.ts [hunk 1]
```diff
@@ -107,11 +107,8 @@ export const usePostModalNavigation = (
       return null;
     }
 
-    const routeHandler = (newRoute: string) => {
-      if (newRoute.includes('/posts/')) {
-        return;
-      }
-      onCloseModal();
+    const routeHandler = () => {
+      onCloseModal(true);
```
Comment (capJavert) [inline packages/shared/src/hooks/usePostModalNavigation.ts:111]: Even though I put true for `fromPopState` basically what it does:
- skips calling `changeHistory`
  - when this `routerHandler` is called Next.js always takes over routing to no need to worry about clearing history
- this is never called on our custom modal history changes so no need for `newRoute.includes` from my testing 
- back button now works when you are navigating between multiple tag pages by constantly selecting them eg. /tags/php -> /tags/javascript -> back -> /tags/php (before URL would change but the page would stay the same as if you are on javascript)

**Comment by capJavert** [comment]
Think this fixes it with minimal changes but please test since there is a lot of edge cases.

**Review by rebelchris — APPROVED** [review]
Lets see definitely cleaner 😅

**Comment by rebelchris** [comment]
Oh didnt we add the posts route check for the soft stuff on webapp push?
Where it would flicker sometimes?

**Comment by capJavert** [comment]
> Oh didnt we add the posts route check for the soft stuff on webapp push?
Where it would flicker sometimes?

@rebelchris, actually we discussed here https://github.com/dailydotdev/apps/pull/2983#discussion_r1562425189, it was added as failsafe but it was not needed in the first place from what I see.

**Review by kopancek — APPROVED** [review]
Looks good to me, like the simplification of the code as well 👍

**Review by omBratteng — APPROVED** [review]
